docs/src/internals/rfc-rooting-by-construction.md

```
@@ -0,0 +1,248 @@
+# RFC: rooting by construction
+
+**Status:** proposal. Nothing in this document is implemented.
+**Problem:** [The GC rooting invariant](gc-rooting-invariant.md) — #7154, #7184,
+#7192, #7206, #7211.
```

markdownlint-cli2 (0.23.1)

[warning] No space after hash on atx style heading

(MD018, no-missing-space-atx)